Lib/test/test_httpservers.py

```diff
@@ -1535,6 +1537,197 @@ def test_server_test_ipv4(self, _):
             server.test(ServerClass=mock_server, bind=bind)
             self.assertEqual(mock_server.address_family, socket.AF_INET)
 
+@mock.patch('http.server.test')
+class CommandLineTestCase(unittest.TestCase):
+    def setUp(self):
+        self.default_port = 8000
+        self.default_bind = None
+        self.default_protocol = 'HTTP/1.0'
+        self.default_handler = SimpleHTTPRequestHandler
+        self.default_server = unittest.mock.ANY
+        self.tls_cert = certdata_file('ssl_cert.pem')
+        self.tls_key = certdata_file('ssl_key.pem')
+        self.tls_password = 'somepass'
+        tls_password_file_object = \
+            tempfile.NamedTemporaryFile(mode='w+', delete=False)
+        tls_password_file_object.write(self.tls_password)
+        self.tls_password_file = tls_password_file_object.name
+        tls_password_file_object.close()
+        self.args = {
+            'HandlerClass': self.default_handler,
+            'ServerClass': self.default_server,
+            'protocol': self.default_protocol,
+            'port': self.default_port,
+            'bind': self.default_bind,
+            'tls_cert': None,
+            'tls_key': None,
+            'tls_password': None,
+        }
+        return super().setUp()
+
+    def tearDown(self):
+        if os.path.exists(self.tls_password_file):
+            os.remove(self.tls_password_file)
+        return super().tearDown()
+
+    def get_random_temporary_file_name(self):
+            return os.path.join(tempfile.gettempdir(), os.urandom(16).hex())
+
+    def invoke_httpd(self, *args):
+        output = StringIO()
+        with contextlib.redirect_stdout(output), \
+            contextlib.redirect_stderr(output):
+            server._main(args)
+        return textwrap.dedent(output.getvalue()).strip()
```

Comment:
if we don't care about parsing the output, then we don't need `textwrap`:
```suggestion
        return output.getvalue()
```